.circleci/config.yml

```diff
@@ -9,14 +9,35 @@ executors:
       CXXFLAGS: -DGEOS_INLINE
 
 commands:
-  install_macos_dependencies:
+  mac_deps:
     steps:
       - run: brew install protobuf cmake ccache libtool libspatialite pkg-config luajit curl wget czmq lz4 spatialite-tools unzip
-      - run: sudo python3 -m pip install shapely "conan<2.0.0"
+      - run: source venv/bin/activate && sudo python3 -m pip install shapely "conan<2.0.0"
       - run: git clone https://github.com/kevinkreiser/prime_server --recurse-submodules && cd prime_server && ./autogen.sh && ./configure && make -j8 && make install
-  install_linux_dependencies:
+  mac_preflight:
     steps:
-      - run: sudo bash ./scripts/install-linux-deps.sh
+      - run: python3 -m venv venv
+  linux_deps:
+    steps:
+      - run: |
+            source venv/bin/activate
+            ./scripts/install-linux-deps.sh
+  linux_preflight:
+    steps:
+      - run: |
+            sudo apt-get update --assume-yes
+            env DEBIAN_FRONTEND=noninteractive sudo apt-get install --yes --quiet python3-pip python3-venv git curl
+            python3 -m venv venv
+  check_ci_lint:
+    steps:
+      - run: |
+            source venv/bin/activate
```

Comment:
annoyingly we always have to get into our venv before we do any new step in CI now but whatever... also notice i broke up the stuff into more steps

Reply:
technically we dont have to do it this way, we can simply export some environment var updates to get the virtual env automatically.. i can update that later, dont want to rock the CI boat at the moment

Reply:
not very opinionated here, I'd have just `--break-system-packages` when installing with pip

Reply:
also needs to go into format.sh, I know. I'm fine with that. nothing changed in the world, pip is just getting parental for some reason. so functionally we'd still do the exact same thing as before. I think I already did that in another PR that's in the queue (doesn't mean anything, just underlines my position:))

Reply:
if you like i can simplify all this by just doing the environment variables to use the virtual env. ill do that quickly :smile: